tests/test_admin.py

```diff
@@ -248,6 +263,147 @@ def test_advanced_settings_link(self):
+            )
+        pagecontent.page.refresh_from_db()
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(
+            response.content.decode("utf-8"), "The page is not eligible to be home."
+        )
+        self.assertFalse(pagecontent.page.is_home)
+
+    def test_non_existing_page_should_result_404(self):
+        with self.login_user_context(self.get_superuser()):
+            response = self.client.post(
+                self.get_admin_url(PageContent, "set_home_content", 99)
+            )
+        self.assertEqual(response.status_code, 404)
+
+    def test_permission_to_set_home_page(self):
+        version = PageVersionFactory(content__page__node__depth=1, state=PUBLISHED)
+        pagecontent = version.content
+        with self.login_user_context(self.get_superuser()), patch(
+            "djangocms_versioning.admin.VersioningAdminMixin.has_change_permission",
+            return_value=False,
+        ) as mock:
+            response = self.client.post(
+                self.get_admin_url(PageContent, "set_home_content", pagecontent.pk)
+            )
+        pagecontent.page.refresh_from_db()
+        self.assertEqual(mock.call_count, 1)
+        self.assertEqual(response.status_code, 403)
+        self.assertFalse(pagecontent.page.is_home)
+
+    def test_when_new_homepage_tree_has_apphooks_should_trigger_signal(self):
+        PageVersionFactory(
+            content__page__node__depth=1, content__page__is_home=1, state=PUBLISHED
+        )
+        version_2 = PageVersionFactory(content__page__node__depth=1, state=PUBLISHED)
+        to_be_homepage = version_2.content
+        with self.login_user_context(self.get_superuser()), patch(
+            "cms.models.query.PageQuerySet.has_apphooks", return_value=True
+        ), patch(
+            "djangocms_pageadmin.admin.set_restart_trigger", return_value=True
+        ) as mock_handler:
+            self.client.post(
+                self.get_admin_url(PageContent, "set_home_content", to_be_homepage.pk)
+            )
+            mock_handler.assert_called()
+
+    def test_when_old_homepage_tree_has_no_apphooks_shouldnt_trigger_signal(self):
+        PageVersionFactory(
+            content__page__node__depth=1, content__page__is_home=1, state=PUBLISHED
+        )
+        version_2 = PageVersionFactory(content__page__node__depth=1, state=PUBLISHED)
+        to_be_homepage = version_2.content
+        with self.login_user_context(self.get_superuser()), patch(
+            "cms.models.query.PageQuerySet.has_apphooks", return_value=False
+        ) as mock, patch(
+            "djangocms_pageadmin.admin.set_restart_trigger"
+        ) as mock_handler:
+            self.client.post(
+                self.get_admin_url(PageContent, "set_home_content", to_be_homepage.pk)
+            )
+
+            mock_handler.assert_not_called()
+
+    def test_when_old_home_tree_is_none_should_not_trigger_signal(self):
```

Comment:
How does this test set old_home_tree to None though? From what I can see you're still mocking `has_apphooks` which would only be called if old_home_tree is not None.